Platform independent core

Ingrim4 opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->

## Description
Adds a platform independent core as basis for platform support other than bukkit e.g. fabric.

## Related Issue
<!--- This project only accepts pull requests related to open issues -->
<!--- If suggesting a new feature or change, please discuss it in an issue first -->
<!--- If fixing a bug, there should be an issue describing it with steps to reproduce -->
<!--- Please link to the issue here: -->

## Motivation and Context
<!--- Why is this change required? What problem does it solve? -->

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->

## Types of Changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

Ingrim4 committed:
chore: move config to new core package

Ingrim4 committed:
chore: rename files so git can track changes on manual merge

Ingrim4 committed:
merge: orebfuscator-next to platform-independent-core

Ingrim4 committed:
feat: wip - add block tag support

Ingrim4 committed:
feat: add block tag support

Ingrim4 committed:
ci: fix most build warnings

NgLoader commented on orebfuscator-core/src/main/java/dev/imprex/orebfuscator/config/OrebfuscatorCacheConfig.java:
```diff
@@ -0,0 +1,165 @@
+    section.set("memoryCache.expireAfterAccess", this.expireAfterAccess);
+
+    section.set("diskCache.enabled", this.enableDiskCacheValue);
+    section.set("diskCache.directory", this.baseDirectory.toString());
```
relativize?

Ingrim4 replied:
Done

NgLoader commented on orebfuscator-core/src/main/java/dev/imprex/orebfuscator/config/OrebfuscatorProximityConfig.java:
```diff
@@ -0,0 +1,198 @@
+      int blockFlags = this.defaultBlockFlags;
+
+      // parse block specific height condition
+      if (block.isInt("minY") && block.isInt("maxY")) {
```
isNumber?

Ingrim4 replied:
Done

NgLoader commented on orebfuscator-core/src/main/java/dev/imprex/orebfuscator/reflect/predicate/RequirementCollector.java:
```diff
@@ -0,0 +1,35 @@
+  private final StringBuilder builder;
+
+  public RequirementCollector(@NotNull String prefix) {
+    builder = new StringBuilder(prefix).append("{\n");
```
use System.lineSeparator()

Ingrim4 replied:
stay on `\n`

NgLoader commented on orebfuscator-core/src/test/java/dev/imprex/orebfuscator/config/yaml/ConfigurationSectionTest.java:
```diff
@@ -0,0 +1,471 @@
+    section.set("number", 1);
+    section.set("other", true);
+
+    assertEquals(1L, section.getDouble("double"));
```
1d instead of 1L

Ingrim4 replied:
Done

Ingrim4 committed:
feat: use vose's alias method for WeightedRandom
feat: migrate BlockPos & NamespacedKey to records
test: add probabilistic unit tests for new WeightedRandom
feat: move state management outside of nms for region file cache

Ingrim4 committed:
feat: support modded blocks

Ingrim4 committed:
feat: improve performance of packet size stats

Ingrim4 committed:
chore: RollingAverage formatting

Ingrim4 committed:
chore: use groupId ('dev.imprex.orebfuscator') for core

Ingrim4 committed:
chore: minor core config cleanup

Ingrim4 committed:
feat: dynamic default config lookup based on game version

Ingrim4 committed:
feat: move chunk package to core module
